tools/e2e-commons/README.md

```diff
@@ -0,0 +1,234 @@
+
+To bypass the offline mode you will need your site to have a publicly accessible url that will proxy all requests to your locally running WordPress installation.
+We use `localtunnel` library to expose `localhost:8889` via a public url.
+
+```shell
+## Decrypt default config file
+CONFIG_KEY=secret_key openssl enc -md sha1 -aes-256-cbc -d -pass env:CONFIG_KEY -in ./node_modules/jetpack-e2e-commons/config/encrypted.enc -out config/local-test.js
+
+## Start and the Docker environment and configure the WordPress installation
+pnpm e2e-env start
+
+## Create a tunnel
+pnpm tunnel on 
+```
+
+The tunnel url will be stored in a file in the config folder of your tests, so that it can be read by the tests and then reused by the tunnel script. See config files for details.
+
+### 1.4. Run the tests
+
+```shell
+NODE_CONFIG_DIR='./config' pnpm jest
+```
+
+## 2. Use a remote preconfigured site
+
+### 2.1. Add the test site details in the config file
+
+Edit the encrypted config file and add an entry in the `testSites` object with the details of your test site
+
+```js
+mySite: {
+    url: 'site-url', 
+    username: 'username',
+    password: 'password',
+    dotComAccount: ['username', 'password']
+}
+```
+
+### 2.2. Run the tests
+
+Set the `TEST_SITE` environment variable with the name of the previously defined configuration object.
+
+```shell
+TEST_SITE=mySite NODE_CONFIG_DIR='./config' pnpm jest
+```
+
+# Functionality plugins
+
+Tests rely on functionality plugins that provide some additional functionality, provide shortcuts, etc.
+
+### e2e-plan-data-interceptor.php
+
+The purpose of this plugin is to provide a way to `mock` Jetpack plan, for cases when we test functionality that does not directly use paid services. Great example of this purpose is a paid Gutenberg blocks.
+
+# Test architecture
+
+## Pages
+
+The tests are using the `PageObject` pattern, which is a way to separate test logic from implementation. Page objects are basically abstractions around specific pages and page components.
+There are two base classes that should be extended by page objects: [`WpPage`](./pages/wp-page.js) and [`PageActions`](./pages/page-actions.js) class.
+
+`WpPage` implements common page methods, like `init` - static method that initializes a page object and checks the displayed page is the expected one, and `visit` - method that navigates to a page URL and then performs all the `init` checks.
+
+`WpPage` extends `PageActions`.
+`WpPage` should be extended by all page objects that represent full pages. Rule of thumb: if it has a URL it should extend WpPage. Otherwise, it's probably representing a page component (like a block) and should directly extend `PageActions`.
+
+Since most of the Playwright functionality is `async`, and JavaScript constructors are not - we should initialize pages with `init()` static method: `await BlockEditorPage.init( page )` to make sure we would wait for `expectedSelectors` checks.
```

Comment:
```suggestion
Since most of the Playwright functionality is `async` - and JavaScript constructors are not - we should initialize pages with the `init()` static method: `await BlockEditorPage.init( page )` to make sure we would wait for `expectedSelectors` checks.
```
or
```suggestion
Since most of the Playwright functionality is `async` and JavaScript constructors are not, we should initialize pages with the `init()` static method: `await BlockEditorPage.init( page )` to make sure we would wait for `expectedSelectors` checks.
```